examples/jsm/renderers/CSS2DRenderer.js

```diff
@@ -135,13 +135,18 @@ class CSS2DRenderer {
 				return;
 
 			}
-			
+
 			if ( object.isCSS2DObject ) {
 
 				_vector.setFromMatrixPosition( object.matrixWorld );
 				_vector.applyMatrix4( _viewProjectionMatrix );
 
-				const visible = ( _vector.z >= - 1 && _vector.z <= 1 ) && ( object.layers.test( camera.layers ) === true );
+				// Cull if outside NDC [-1, +1]
+				if ( _vector.z >= - 1 && _vector.z <= 1 ) return;
```

Comment:
I'm afraid this early out isn't right. You need to store the result in a `visible` property so it is possible to configure the `element.style.display` a few lines below.

Reply:
I might just prefer to not early out anywhere in this PR.